src/main/kotlin/by/architecture_map/belarus/service/impl/ConstructionServiceImpl.kt

```diff
@@ -12,40 +12,53 @@ class ConstructionServiceImpl(
 ) : ConstructionService {
 
     override fun create(construction: Construction) = constructionRepository.save(construction)
-
     override fun findAll() = constructionRepository.findAll()
 
     override fun find(id: Int): Construction = constructionRepository.findById(id)
         .orElseThrow { throw NotFoundException("Construction not found with id: $id") }
 
     override fun update(id: Int, updatedConstruction: Construction): Construction =
-        applyUpdates(id) {
-            name = updatedConstruction.name
-            description = updatedConstruction.description
-            architecturalStyle = updatedConstruction.architecturalStyle
-            address = updatedConstruction.address
-            architects = updatedConstruction.architects
-            buildingDate = updatedConstruction.buildingDate
-            images = updatedConstruction.images
-        }
 
-    override fun patchUpdate(id: Int, updatedConstruction: Construction): Construction =
-        applyUpdates(id) {
-            if (!updatedConstruction.name.isNullOrEmpty())
+        constructionRepository.findById(id)
+            .orElseThrow { NotFoundException("Construction not found with id: $id") }
+            .apply {
                 name = updatedConstruction.name
-            if (updatedConstruction.address != null)
+                description = updatedConstruction.description
+                architecturalStyle = updatedConstruction.architecturalStyle
                 address = updatedConstruction.address
-            if (!updatedConstruction.architects.isNullOrEmpty())
                 architects = updatedConstruction.architects
-            if (updatedConstruction.architecturalStyle != null)
-                architecturalStyle = updatedConstruction.architecturalStyle
-            if (!updatedConstruction.images.isNullOrEmpty())
-                images = updatedConstruction.images
-            if (!updatedConstruction.description.isNullOrEmpty())
-                description = updatedConstruction.description
-            if (!updatedConstruction.buildingDate.isNullOrEmpty())
                 buildingDate = updatedConstruction.buildingDate
-        }
+                buildingCentury = updatedConstruction.buildingCentury
+                images = updatedConstruction.images
+            }
+            .let {
+                constructionRepository.save(it)
+            }
+
+    override fun patchUpdate(id: Int, updatedConstruction: Construction): Construction =
+        constructionRepository.findById(id)
```

Comment:
why have you removed the previous `applyUpdates` method or at least do not reuse `fun find(id: Int)`?

Reply:
I'm surprised too. Apparently, I am don't correctly resolved merge conflicts. I am sorry(